cmake/kokkos_enable_options.cmake

```
@@ -197,13 +197,14 @@ if(KOKKOS_COMPILE_LANGUAGE STREQUAL CUDA)
   endif()
 endif()
 
-# This is known to occur with Clang 9. We would need to use nvcc as the linker
+# This is known to occur with Clang 9 until Clang 15. We would need to use nvcc as the linker
 # http://lists.llvm.org/pipermail/cfe-dev/2018-June/058296.html
-# TODO: Through great effort we can use a different linker by hacking
-# CMAKE_CXX_LINK_EXECUTABLE in a future release
-if(KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE AND KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+if(KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE AND KOKKOS_CXX_COMPILER_ID STREQUAL Clang AND KOKKOS_CXX_COMPILER_VERSION
+                                                                                            VERSION_LESS 15
```

```suggestion
                                                                                            VERSION_LESS 18
```
I would be a bit more conservative, if we don't want to test versions less than 18, let's not allow it ? Moreover we don't have CI about that yet.

---

Increased version requirement to Clang 17.